app/controllers/api/v2/TimelinesController.js

```diff
@@ -75,6 +76,10 @@ export const ownTimeline = (feedName, params = {}) =>
         throw new NotFoundException(`Timeline is not found`);
       }
 
+      if (feedName === 'RiverOfNews' && (await user.setFirstInteraction())) {
+        await scheduleWelcomeDirects(user);
+      }
```

Comment:
I worry, that we don't have transaction here.
can we?

Reply:
Do you mean that this code may set the first interaction time, but throw an error in scheduleWelcomeDirects? 
It is, in theory, possible, but what we should do in this case? Rollback the setFirstInteraction and wait for the next homepage load?

Reply:
Yes, exactly. Or just start with auto-retry of transaction